Makefile

```diff
@@ -494,31 +496,37 @@ doc-all-except-pdf:
 	cd doc && $(MAKE) all-except-pdf
 
 # Get the current release version
-ifneq ($(shell ls /cse/web/research/plse/daikon/download/daikon-*.zip 2>/dev/null),)
-    CUR_VER := $(shell ls /cse/web/research/plse/daikon/download/daikon-*.zip |perl -p -e 's/^.*download.daikon.//' |perl -p -e 's/.zip//')
+ifneq ($(shell ls $(WWW_DIR)/download/daikon-*.zip 2>/dev/null),)
+    CUR_VER := $(shell ls $(WWW_DIR)/download/daikon-*.zip |perl -p -e 's/^.*download.daikon.//' |perl -p -e 's/.zip//')
     CUR_RELEASE_NAME := daikon-$(CUR_VER)
-    NEW_RELEASE_NAME := daikon-$(shell cat doc/VERSION)
+    NEW_VER := $(shell cat doc/VERSION)
+    NEW_RELEASE_NAME := daikon-$(NEW_VER)
 else
     CUR_RELEASE_NAME := UNKNOWN
     NEW_RELEASE_NAME := UNKNOWN
 endif
 
 check-for-broken-doc-links:
-	checklink -q -r `grep -v '^#' ${DAIKONDIR}/plume-lib/bin/checklink-args.txt` http://plse.cs.washington.edu/daikon/staging-daikon >check.log 2>&1
+	checklink -q -r `grep -v '^#' ${DAIKONDIR}/plume-lib/bin/checklink-args.txt` http://plse.cs.washington.edu/staging-daikon  >check.log 2>&1
 
-HISTORY_DIR := /cse/web/research/plse/daikon/history
+HISTORY_DIR := $(STAGING_DIR)/history
 save-current-release:
 	@echo Saving $(CUR_VER) to history directory.
 	-chmod u+w $(HISTORY_DIR)
 	mkdir $(HISTORY_DIR)/$(CUR_RELEASE_NAME)
 	-chmod u-w $(HISTORY_DIR)
-	cd $(HISTORY_DIR)/$(CUR_RELEASE_NAME) && cp -p /cse/web/research/plse/daikon/download/$(CUR_RELEASE_NAME).zip . && unzip -p $(CUR_RELEASE_NAME).zip $(CUR_RELEASE_NAME)/doc/CHANGES >CHANGES && chmod o-w CHANGES .
+	cd $(HISTORY_DIR)/$(CUR_RELEASE_NAME) && cp -p $(WWW_DIR)/download/$(CUR_RELEASE_NAME).* . && unzip -p $(CUR_RELEASE_NAME).zip $(CUR_RELEASE_NAME)/doc/CHANGES >CHANGES && chmod o-w CHANGES .
 
 # Perl command compresses multiple spaces to one, for first 9 days of month.
 ifeq ($(origin TODAY), undefined)
 TODAY := $(shell date "+%B %e, %Y" | perl -p -e 's/  / /')
 endif
 
+update-and-commit-version: update-doc-dist-date-and-version
+	git commit -a -m "Change version to $(NEW_VER)"
+	git push
+	cd fjalar && git commit -a -m "Change version to $(NEW_VER)" && git push
```

Comment:
If you don't want to change the working directory, I think you need to enclose this line in parentheses.  (You would need to do that in the shell and I think in a Makefile too.)